cypress/support/commands/element_selectors.js

```diff
@@ -1,57 +1,104 @@
-/* eslint-disable no-undef */
-
 /**
- * Retrieves a form footer button by its name and type.
+ * Retrieves a form footer button by its text content and type using an object parameter.
  *
- * @param {string} name - The name or text content of the button.
- * @param {string} [type='button'] - The HTML button type (e.g., 'button', 'submit', 'reset'). Defaults to 'button'.
+ * @param {Object} options - The options object.
+ * @param {string} options.buttonText - The text content of the button (required).
+ * @param {string} [options.buttonType='button'] - The HTML button type (e.g., 'button', 'submit', 'reset'). Defaults to 'button'.
  * @returns {Element} The matched button element.
+ * @throws {Error} If buttonName is not provided.
  *
  * Example:
- *   cy.getFormFooterButtonByType('Save Changes');
- *   cy.getFormFooterButtonByType('Reset', 'reset');
+ *   cy.getFormFooterButtonByTypeWithText({ buttonText: 'Save Changes' });
+ *   cy.getFormFooterButtonByTypeWithText({ buttonText: 'Submit', buttonType: 'submit' });
  */
-Cypress.Commands.add('getFormFooterButtonByType', (name, type = 'button') =>
-  cy.contains(`#main-content .bx--btn-set button[type="${type}"]`, name)
+Cypress.Commands.add(
+  'getFormFooterButtonByTypeWithText',
+  ({ buttonType = 'button', buttonText } = {}) => {
+    if (!buttonText) {
+      cy.logAndThrowError('buttonName is required');
```

Comment:
As a follow up, we call the function WithText, perhaps buttonText would be a better term to use in the error message.